src/main/cpp/src/utils.hpp

```diff
@@ -0,0 +1,155 @@
+/*
```

Comment:
Oh what `utils` in the file name? The code seems date/time related. We already have `datetime_utils.hpp` for the function declaration, but here we have device code thus it should be something like `datetime_utils.cuh`.